yarn-project/aztec-rpc/src/kernel_prover/kernel_prover.test.ts

```diff
@@ -69,9 +69,14 @@ describe('Kernel Prover', () => {
   };
 
   const expectExecution = (fns: string[]) => {
-    const callStackItems = proofCreator.createProof.mock.calls.map(args => args[2].callStackItem.functionData);
-    expect(callStackItems).toEqual(fns);
-    proofCreator.createProof.mockClear();
+    const callStackItemsInit = proofCreator.createProofInit.mock.calls.map(args => args[1].callStackItem.functionData);
+    const callStackItemsInner = proofCreator.createProofInner.mock.calls.map(
+      args => args[1].callStackItem.functionData,
+    );
+
```

Comment:
It will be great to check that both methods are called the correct number of times. In case that someone accidentally changes something and result in `createProofInit` being called more than once. We know that there will always be 1 init and followed by inner(s), so we can test it this way:

```js
expect(proofCreator.createProofInit).toHaveBeenCalledTimes(1);
expect(proofCreator.createProofInner).toHaveBeenCalledTimes(fns.length - 1);
```